### ISSUE TITLE
Upstream prematurely closed connection while reading response header from upstream

### ISSUE THREAD

**Issue by lsl** [Automattic/wp-calypso#61434]
I caught this error today while loading https://wordpress.com on my phone. The error looks like an nginx error but is being caused by something on the calypso server side of things (I'm pretty sure).

See slack convo for more context and pastebin for logs. It seems to be nothing new but relatively rare. It only happens on the /log-in route and looks like it might be some kind of memory issue (e.g. maybe [docker is killing the nodejs server process](https://bobcares.com/blog/nginx-docker-error-16440-upstream-prematurely-closed-connection/))

p1645676625677859-slack-C02AVAR9B / 2bc73-pb (logs)

```
[error] 22757#0: *5083248 upstream prematurely closed connection while reading response header from upstream, "GET /log-in HTTP/1.1", upstream: "http://127.0.0.1:3004/log-in", host: "wordpress.com"
```

I didn't screenshot the error today but it's been reported before (potentially same issue, just less frequent) and a screenshot matching what I saw was listed there p1526766761000066-slack-C02AVAR9B / p4TIVU-8Vt-p2

![resetconnection](https://user-images.githubusercontent.com/811776/155466488-c02e2e98-abf3-499b-97f1-e9f43ccbb93c.png)

**Comment by noahtallen** [comment]
cc @Automattic/team-calypso-frameworks

**Comment by tyxla** [comment]
Unfortunately, I don't have much more than the logs you already shared. I'd suggest escalating it to systems in case they have some additional information we don't have access to.

**Comment by lsl** [comment]
I don't think they have access to much more than what I linked above. This isn't a pressing issue but it does seem like an ongoing one. Lets keep this open for now & see if we get further reports from HE's

**Comment by tyxla** [comment]
I've been trying to reproduce it, but with no luck. I wonder if you have any observations about whether it happens more often under certain circumstances.

**Comment by lsl** [comment]
No, I've only seen it once and it's worth noting I don't think my request was the cause. Someone else's request seems to cause some kind of problem at the calypso server level which then effects other requests. (pretty sure about this anyway)

**Comment by jordesign** [comment]
Cloising this as a one-off issue with no further reports in the past (almost) 3 years.
